algopy_testing/src/algopy/op.py

```diff
@@ -168,3 +178,398 @@ def vrf_verify(
+            # turn the pairs into the dictionay for the top level,
+            # all other levels remain as key value pairs
+            # e.g.
+            # input bytes: b'{"key0": 1,"key1": {"key2":2,"key2":"10"}, "key2": "test"}'
+            # output dict: {'key0': 1, 'key1': [('key2', 2), ('key2', '10')], 'key2': 'test'}
+            result = dict(pairs)
+            if len(pairs) != len(result):
+                raise ValueError(
+                    "error while parsing JSON text, invalid json text, duplicate keys found"
+                )
+            return result  # noqa: TRY300
+        except json.JSONDecodeError:
+            raise ValueError("error while parsing JSON text, invalid json text") from None
+
+    @staticmethod
+    def _raise_key_error(key: str) -> None:
+        raise ValueError(f"key {key} not found in JSON text")
+
+    @staticmethod
+    def json_string(a: Bytes | bytes, b: Bytes | bytes, /) -> Bytes:
+        b_str = _bytes_to_string(b, "can't decode bytes as string")
+        obj = JsonRef._load_json(a)
+        try:
+            result = obj[b_str]
+        except KeyError:
+            JsonRef._raise_key_error(b_str)
+
+        if not isinstance(result, str):
+            raise TypeError(f"value must be a string type, not {type(result).__name__!r}")
+
+        # encode with `surrogatepass` to allow sequences such as `\uD800`
+        # decode with `replace` to replace with official replacement character `U+FFFD`
+        # encode with default settings to get the final bytes result
+        result = result.encode("utf-16", "surrogatepass").decode("utf-16", "replace").encode()
+        return Bytes(result)
+
+    @staticmethod
+    def json_uint64(a: Bytes | bytes, b: Bytes | bytes, /) -> UInt64:
+        b_str = _bytes_to_string(b, "can't decode bytes as string")
+        obj = JsonRef._load_json(a)
+        try:
+            result = obj[b_str]
+        except KeyError:
+            JsonRef._raise_key_error(b_str)
+
+        result = as_int(result, max=MAX_UINT64)
+        return UInt64(result)
+
+    @staticmethod
+    def json_object(a: Bytes | bytes, b: Bytes | bytes, /) -> Bytes:
+        b_str = _bytes_to_string(b, "can't decode bytes as string")
+        obj = JsonRef._load_json(a)
+        try:
+            # using a custom dict object to allow duplicate keys which is essentially a list
+            result = obj[b_str]
+        except KeyError:
+            JsonRef._raise_key_error(b_str)
+
+        if not isinstance(result, list) or not all(isinstance(i, tuple) for i in result):
+            raise TypeError(f"value must be an object type, not {type(result).__name__!r}")
+
+        result = _MultiKeyDict(result)
+        result_string = json.dumps(result, separators=(",", ":"))
+        return Bytes(result_string.encode())
+
+
+class _MultiKeyDict(dict[Any, Any]):
+    def __init__(self, items: list[Any]):
+        self["something"] = "something"
```

Comment:
Is this meant to be here?

Reply:
I got it off [this stackoverflow answer](https://stackoverflow.com/a/29520802). Without that line, we get an empty object when serialising.